tls/s2n_security_policies.h

```diff
@@ -27,10 +27,32 @@
 
 struct s2n_security_policy {
     uint8_t minimum_protocol_version;
+    /* TLS 1.0 - 1.2 - cipher preference includes multiple elements such
+     * as signature algorithms, record algorithms, and key exchange algorithms
+     * TLS 1.3 - cipher preference only determines record encryption
+     */
     const struct s2n_cipher_preferences *cipher_preferences;
+    /* kem_preferences is only used for Post-Quantum cryptography */
     const struct s2n_kem_preferences *kem_preferences;
+    /* TLS 1.2 - optional extension to specify signature algorithms other than
+     * default: https://www.rfc-editor.org/rfc/rfc5246#section-7.4.1.4.1
+     * TLS 1.3 - required extension specifying signature algorithms
```

Comment:
I guess it's kind of odd to refer to the extension for any of these. This list is used by the client to build the extension, but also by the server when choosing one of the options from the extension.

Reply:
I tried to be more explicit about this in the next revision, but I do think it is useful to refer to the extension, because is then unlocks a wealth of documentation that we don't necessarily have in our own codebase. E.g. if I want to learn more about `signature_preferences`, a good starting point it looking up information about the `signature_algorithms_cert`.